bpf-rust-programs/prediction-poll/program_data/src/tally.rs

```diff
@@ -53,16 +53,17 @@ impl TallyData<'_> {
         *self.tally_count as usize
     }
 
-    pub fn add_tally(&mut self, user_key: &SolPubkey, wager: u64) {
-        self.tallies[self.len()][..32].copy_from_slice(user_key);
-        self.tallies[self.len()][32..].copy_from_slice(&wager.to_le_bytes());
+    pub fn add_tally(&mut self, user_key: &Pubkey, wager: u64) {
+        let next_tally = self.len();
+        self.tallies[next_tally][..32].copy_from_slice(user_key.as_ref());
+        self.tallies[next_tally][32..].copy_from_slice(&wager.to_le_bytes());
         *self.tally_count += 1;
     }
 
-    pub fn iter(&self) -> impl Iterator<Item = (&SolPubkey, u64)> {
-        self.tallies[..self.len()].iter().map(|t| {
-            let key = array_ref!(t, 0, 32);
-            let wager = u64::from_le_bytes(*array_ref!(t, 32, 8));
+    pub fn iter(&self) -> impl Iterator<Item = (Pubkey, u64)> + '_ {
```

Comment:
@jackcmay needed the super intuitive syntax of `+ '_` to get passed that lifetime issue 🤷‍♂